paddle/contrib/inference/paddle_inference_api.h

```diff
@@ -12,49 +12,58 @@
    See the License for the specific language governing permissions and
    limitations under the License. */
 
+/*
+ * This file contains the definition of a simple Inference API for Paddle.
+ *
+ * ATTENTION: It requires some C++ features, for lower version C++ or C, we
+ * might release another API.
+ */
+
 #pragma once
 
+#include <memory>
 #include <string>
 #include <vector>
 
-namespace paddle {
+struct PaddleTensor {
+  std::string name;  // variable name.
+  std::vector<int> shape;
+  std::vector<unsigned char> data;         // bytes of data.
+  size_t type{typeid(float).hash_code()};  // hash of type
+};
 
-class Predictor {
+/*
+ * A simple Inference API for Paddle. Currently this API might just be used by
+ * non-sequence scenerios.
+ * TODO(Superjomn) Prepare another API for NLP-related usages.
+ */
+class PaddlePredictor {
 public:
-  struct Attr;
-  Predictor() = default;
-
-  // Build the network before inference.
-  bool Init(const Attr& attr);
+  struct Config;
+  PaddlePredictor() = default;
 
   // Predict an record.
-  // Arguments:
-  //   inputs: the name of the input variables.
-  //   outputs: the name of the output varaibles.
-  //   input_shapes: the shape of the input variables.
-  //   output_shapes: the shape of the output variables.
-  //   input_data: the data of the input variables.
-  //   output_data: the data of the output variables.
-  bool Run(const std::vector<std::string>& inputs,
-           const std::vector<std::string>& outputs,
-           const std::vector<std::vector<int>>& input_shapes,
-           const std::vector<std::vector<int>>& output_shapes,
-           const std::vector<std::vector<float>>& input_data,
-           std::vector<std::vector<float>>* output_data);
-
-  // Clone a predictor that share the model weights.
-  Predictor* Clone();
+  virtual bool Run(const std::vector<PaddleTensor>& inputs,
+                   std::vector<PaddleTensor>* output_data) = 0;
+
+  // Clone a predictor that share the model weights, the Cloned predictor should
+  // be thread-safe.
+  virtual std::unique_ptr<PaddlePredictor> Clone() = 0;
```

Comment:
If it's not required now, I suggest we leave it for future...Clone can easily be a source of bugs.

Reply:
Currently, just return `nullptr`, this API is required by the image team.

Reply:
I see.